lib/instrumentation/http-shared.js

```diff
@@ -207,7 +207,10 @@ exports.traceOutgoingRequest = function (agent, moduleName, method) {
   return function wrapHttpRequest(orig) {
     return function wrappedHttpRequest(input, options, cb) {
       const parentRunContext = ins.currRunContext();
-      var span = ins.createSpan(null, 'external', 'http', { exitSpan: true });
+      var span =
+        input?.headers?.['x-amz-invocation-type'] === 'RequestResponse'
+          ? null
+          : ins.createSpan(null, 'external', 'http', { exitSpan: true });
```

Comment:
Was this change to avoid the extra span necessary for the Service Map connection to work? Or was this just to avoid having an addition span in the trace waterfall?

Reply:
Both I guess. If this span is created when a lambda is invoked, I can see two entries on Service Map: one empty (created by this span) and other with the full tracing (created from the code send in `lib/lambda.js`.

This code just removes the empty one.